test/functional/publications_controller_test.rb

```diff
@@ -217,7 +217,7 @@ def assert_featured(doc)
   end
 
   view_test "#index orders consultations by first_published_at date by default" do
-    consultations = 5.times.map {|i| create(:published_consultation, opening_at: (10 - i).days.ago) }
+    consultations = 5.times.map {|i| create(:published_consultation, first_published_at: (10 - i).days.ago) }
```

Comment:
[The UI](https://www.gov.uk/government/publications) allows you to refine the search by "Published after" and "Published before" which is exactly the behaviour this PR is now implementing correctly.

<img width="232" alt="screen shot 2017-01-16 at 20 18 48" src="https://cloud.githubusercontent.com/assets/2715/21997827/2ed118ea-dc29-11e6-8542-bf5775354b4c.png">

The change in the test setup is needed because the `first_published_at` value will otherwise get set on save unless it is overridden when using the factory.

I've looked into how the search ordering works and I think the only way to implement the behaviour you describe would be to override the `first_public_at` method from `Edition` which is normally:

```rb
def first_public_at
  first_published_at
end
```

That doesn't make sense to me as the value returned would be unexpected.